tests/org.eclipse.reddeer.eclipse.test/src/org/eclipse/reddeer/eclipse/test/jdt/ui/wizards/NewInterfaceCreationWizardTest.java

```diff
@@ -0,0 +1,199 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Red Hat, Inc and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Red Hat, Inc - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.reddeer.eclipse.test.jdt.ui.wizards;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.reddeer.common.exception.RedDeerException;
+import org.eclipse.reddeer.eclipse.jdt.ui.wizards.JavaProjectWizard;
+import org.eclipse.reddeer.eclipse.jdt.ui.wizards.NewInterfaceCreationWizard;
+import org.eclipse.reddeer.eclipse.jdt.ui.wizards.NewInterfaceCreationWizardPage;
+import org.eclipse.reddeer.eclipse.jdt.ui.wizards.NewJavaProjectWizardPageOne;
+import org.eclipse.reddeer.eclipse.ui.perspectives.JavaPerspective;
+import org.eclipse.reddeer.junit.runner.RedDeerSuite;
+import org.eclipse.reddeer.requirements.openperspective.OpenPerspectiveRequirement.OpenPerspective;
+import org.eclipse.reddeer.swt.impl.button.PushButton;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Test class for Java Interface Wizard
+ * 
+ * @author zcervink@redhat.com
+ * 
+ */
+@RunWith(RedDeerSuite.class)
+@OpenPerspective(JavaPerspective.class)
+public class NewInterfaceCreationWizardTest {
+
+	protected static final String PROJECT_NAME = "testproject";
+	protected static final String INTERFACE_NAME = "InterfaceName";
+	protected static final String SOURCE_FOLDER_NAME = PROJECT_NAME + "/src";
+	protected static final String PACKAGE = "org.example.test";
+	protected static final String ENCLOSING_TYPE_CUSTOM_VALUE = "org.example.test." + INTERFACE_NAME;
+
+	@BeforeClass
+	public static void setup() {
+		createTestProject();
+	}
+
+	@Test
+	public void testCreateNewInterface() {
+		// open wizard and fill the interface name
+		NewInterfaceCreationWizard interfaceDialog = new NewInterfaceCreationWizard();
+		interfaceDialog.open();
+		NewInterfaceCreationWizardPage interfacePage = new NewInterfaceCreationWizardPage(interfaceDialog);
+		interfacePage.setName(INTERFACE_NAME);
+
+		// finish the wizard
+		interfaceDialog.finish();
+	}
+
+	@Test
+	public void testSettingAndGettingInputAndCheckboxValues() {
+		// open the wizard
+		NewInterfaceCreationWizard interfaceDialog = new NewInterfaceCreationWizard();
+		interfaceDialog.open();
+		NewInterfaceCreationWizardPage interfacePage = new NewInterfaceCreationWizardPage(interfaceDialog);
+
+		// fill and test values in input fields
+		interfacePage.setName(INTERFACE_NAME);
```

Comment:
This is just minor. But if you are going to edit/update the PR again, could you please use that nice feature of set methods that returns this object and call all set actions in one chain, aka interfacePage.setName("bla").setPackage("bla2").etc... ?

Reply:
Fixed. It really looks better and more compact.